pkg/apis/certmanager/v1/generic_issuer.go

```diff
@@ -0,0 +1,85 @@
+/*
```

Comment:
I added this back in to v1 because there are bunch of controllers that still use it.
It was removed in response to one of my code review comments  in https://github.com/jetstack/cert-manager/pull/3038/#discussion_r449145480 and at that time @munnerz  wrote:

> This is used by the 'issuers' and 'clusterissuers' controllers so that they can work with both Issuer types and ClusterIssuer (by having them both implement 'GenericIssuer').
> 
> The signatures returned version types (e.g. `GetSpec()` returns a _v1beta1_ IssuerSpec etc).
> 
> It's not actually used right now in this API version, so we can remove it. I agree it shouldn't be in this package, and I think it _should_ be possible to 'refactor this away' with some better written controllers for issuers.
> 
> Will remove from v1beta1, but leaving in v1alpha3 and 2 as I don't want to make a retroactive change in this PR smile

I considered doing that refactoring here, but decided it would cause too much code churn and I'm not really familiar enough with the code to do a good job so close to the v1 release. 

See https://github.com/jetstack/cert-manager/issues/3200